proxy_ring_buffer.go

```diff
@@ -0,0 +1,278 @@
+	}
+	return rb.convertToObject(codec.DecodeRingbufferReadOneResponse(response))
+}
+
+// Capacity returns the capacity of this Ringbuffer.
+func (rb *Ringbuffer) Capacity(ctx context.Context) (int64, error) {
+	if rb.capacity == ReadResultSetSequenceUnavailable {
+		request := codec.EncodeRingbufferCapacityRequest(rb.name)
+		response, err := rb.invokeOnPartition(ctx, request, rb.partitionID)
+		if err != nil {
+			return ReadResultSetSequenceUnavailable, err
+		}
+		rb.capacity = codec.DecodeRingbufferCapacityResponse(response)
+	}
+	return rb.capacity, nil
+}
+
+// Size returns number of items in the Ringbuffer. If no ttl is set, the size will always be equal to capacity after the
+// head completed the first loop-around the ring. This is because no items are getting retired.
+func (rb *Ringbuffer) Size(ctx context.Context) (int64, error) {
+	request := codec.EncodeRingbufferSizeRequest(rb.name)
+	response, err := rb.invokeOnPartition(ctx, request, rb.partitionID)
+	if err != nil {
+		return ReadResultSetSequenceUnavailable, err
+	}
+	return codec.DecodeRingbufferSizeResponse(response), nil
+}
+
+// TailSequence returns the sequence of the tail. The tail is the side of the ringbuffer where the items are added to.
+// The initial value of the tail is -1.
+func (rb *Ringbuffer) TailSequence(ctx context.Context) (int64, error) {
+	request := codec.EncodeRingbufferTailSequenceRequest(rb.name)
+	response, err := rb.invokeOnPartition(ctx, request, rb.partitionID)
+	if err != nil {
+		return ReadResultSetSequenceUnavailable, err
+	}
+	return codec.DecodeRingbufferTailSequenceResponse(response), nil
+}
+
+// HeadSequence returns the sequence of the head. The head is the side of the ringbuffer where the oldest items in the ringbuffer
+// are found. If the RingBuffer is empty, the head will be one more than the tail.
+// The initial value of the head is 0 (1 more than tail).
+func (rb *Ringbuffer) HeadSequence(ctx context.Context) (int64, error) {
+	request := codec.EncodeRingbufferHeadSequenceRequest(rb.name)
+	response, err := rb.invokeOnPartition(ctx, request, rb.partitionID)
+	if err != nil {
+		return ReadResultSetSequenceUnavailable, err
+	}
+	return codec.DecodeRingbufferHeadSequenceResponse(response), nil
+}
+
+// RemainingCapacity
+// Returns the remaining capacity of the Ringbuffer. The returned value could be stale as soon as it is returned.
+// If ttl is not set, the remaining capacity will always be the capacity.
+func (rb *Ringbuffer) RemainingCapacity(ctx context.Context) (int64, error) {
+	request := codec.EncodeRingbufferRemainingCapacityRequest(rb.name)
+	response, err := rb.invokeOnPartition(ctx, request, rb.partitionID)
+	if err != nil {
+		return ReadResultSetSequenceUnavailable, err
+	}
+	return codec.DecodeRingbufferRemainingCapacityResponse(response), nil
+}
+
+// ReadMany
+// Reads a batch of items from the Ringbuffer. If the number of available items after the first read item is smaller
```

Comment:
Similar comment about whitespacing in apidocumentation. I think in the IDE it does not look good in when it is one paragraph:

![image](https://user-images.githubusercontent.com/32355782/174244460-b49b20bb-add0-4c30-a477-ad540a093c91.png)

If this is required in go, ignore me

Reply:
🤷 I did copy the documentation from
https://github.com/hazelcast/hazelcast-client-protocol/blob/60185af4a914b0a4742b9b83e1d6ff9f27762057/protocol-definitions/Ringbuffer.yaml#L244

Would it make sense to patch the reference first?

(see also Slack).